src/Http/Controllers/VoyagerProfileController.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace TCG\Voyager\Http\Controllers;
+
+use Illuminate\Http\Request;
+use TCG\Voyager\Events\BreadDataUpdated;
+use TCG\Voyager\Facades\Voyager;
+
+class VoyagerProfileController extends Controller
+{
+    public function index()
+    {
+        return Voyager::view('voyager::profile');
+    }
+
+    // POST BR(E)AD
+    public function update(Request $request)
+    {
+        $slug = 'users';
+
+        $dataType = Voyager::model('DataType')->where('slug', '=', $slug)->first();
+
+        $id = $request->user()->id;
+
+        $data = call_user_func([$dataType->model_name, 'findOrFail'], $id);
+
+        // Check permission
+        $this->authorize('edit', $data);
+
+        //Verify if user tries to modify his own role
+        if ($request->has('role_id')) {
+            $request->merge(['role_id' => $request->user()->role->id]);
+        }
+
+        if($request->has('user_belongstomany_role_relationship')){
+            $request->offsetUnset('user_belongstomany_role_relationship');
+        }
+
+        // Validate fields with ajax
+        $val = $this->validateBread($request->all(), $dataType->editRows, $slug, $id);
+
+        if ($val->fails()) {
+            return response()->json(['errors' => $val->messages()]);
+        }
+
+        if (!$request->ajax()) {
+            $this->insertUpdateData($request, $slug, $dataType->editRows, $data);
+
+            event(new BreadDataUpdated($dataType, $data));
+            
+            $route = $request->user()->can('browse', $data) ? 'voyager.users.index' : 'voyager.profile';
```

Comment:
Can we not just always redirect the user to his profile page?

Reply:
I think it can be done, if we exclude saving it in session, adding a returnUrl hidden input in edit page, so you know if you are editing coming from profile or user list.

I liked the more generic idea to have user sent back as @carlost89 suggested in #2625 :
```php
return redirect()->back()
```
when they don't have browse permission, but that would need to be applied in VoyagerBase.

Reply:
@MrCrayon suggestion looks good:

```php
return redirect()->back();
```

Reply:
You should never reach the point where you don't know where to redirect back to with this.  You can view your profile, edit your profile, and save your edited profile.  We just need to choose whether you go back to the "edit" form page, or the "view" page.  My preference for the view page.  Using `redirect()->back()` solves the problem to an extent as it will always go back to the same place since there's only one form that should ever submit to this, right?

I agree we don't need the conditional here, but I don't quite agree with `redirect()->back()` either.  Convince me?

Reply:
@fletch3555 we could decouple the relation between `/admin/profile` and `/admin/users/1`?

Yes we have to create two controllers, but I don't see any problems.

### My Profile feature

View URL: `/admin/profile`
Edit URL: `/admin/profile/edit` (redirects to: `/admin/profile`)

### User BREAD feature

Browse URL: `/admin/users`
View URL: `/admin/users/{id}`
View URL: `/admin/users/{id}` (redirects to: `/admin/users/{id}/edit`)

Reply:
Sorry for the delay, but yes @cyrildewit, that's exactly what this PR does.

Reply:
@fletch3555 not exactly. If you edit your profile at the moment, you will be pointed to `/admin/users/{id}/edit` instead of `/admin/profile/edit`.

Reply:
That's not what this PR should be doing, though I haven't looked at the code in a while

Reply:
Do you mean that this PR should not add the extra route `/admin/profile/edit`?

Reply:
I was talking with @emptynick yesterday about this feat. I'll do some improvements with the routes to use `admin/profile/edit` when accessing the profile edit page and `admin/user/{id}/edit` when is editing an user.
I'll do it on this week!

@fletch3555 is there anything you want me to improve on the profile feature?